File: DSA Implementations/Python/linear_search.py
def linear_search(arr,k):
    for i in range(len(arr)):
        if arr[i] == k:
            return i
    return -1

n = int(input("Enter the number of elements in the array : "))
arr = []

for i in range(n):
    arr.append(int(input(f"Enter element {i+1} : ")))

k =  int(input("Enter the element to be searched : "))

linear_search_result = linear_search(arr, k)

if linear_search_result != -1:
    print(f"Element {k} found at index {linear_search_result}.")
else:
    print(f"Element {k} not found in the array.")

File: DSA Implementations/C/linear_search.c
#include <stdio.h>

int linear_search(int arr[], int n, int k)
{
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == k)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    int n, k;

    printf("Enter the number of elements in the array : ");
    scanf("%d", &n);

    int arr[n];

    printf("Enter the elements of the array : \n");

    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }

    printf("Enter the element to be searched : ");
    scanf("%d", &k);

    int linear_search_result = linear_search(arr, n, k);

    if (linear_search_result != -1)
    {
        printf("Element found at index %d\n", linear_search_result);
        return 0;
    }
    else
    {
        printf("Element not found in the array\n");
        return 1;
    }

    printf("Element not found in the array\n");
    return 0;
}